### PR TITLE
[Protocol3] Implement UniswapTokenSeller contract

### PR DESCRIPTION
**Pull request by dong77**

### CODE REVIEW

File: packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    function getSlippage(
+        uint amountB,
+        uint amountB2
+        )
+        private
+        pure
+        returns (uint8)
+    {
+        require(amountB > 0 && amountB2 > 0, "INVALID_PRICE");
+        uint slippageBips = amountB.mul(2).sub(amountB2).mul(10000) / amountB;
```
[the remark this answers is not in the document]
Comment (dong77) [reply packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol]: The algorithm works as follows: give`x` tokenS to sell, we ask uniswap how much tokenB we can buy, as `y`. Then we ask again if we sell `2x`, how much we can get from uniswap, as `y'`. We know `y' <= 2*y` for sure. Then we use `(2y - y')/y` to measure the slippage of price. If the slippage is smaller than 1 percentage, the depth is believed to be good enough.

The caller will determine what amount to transfer to this contract to sell -- the transfer of tokens/ether and calling sellToken shall be in tx so if selling fails the fund will be returned to the caller. This contract sells all the tokens it has.

[the remark this answers is not in the document]
Comment (dong77) [reply packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol]: 1 basis point is 0.01 percent, so we use `(2y - y')*10000/y` to calculate the slippage_bips and compare with 100 bips.

File: packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            require(
+                token.approve(address(exchange), MAX_UINT),
+                "APPROVAL_FAILURE"
+            );
+        }
+    }
+
+    function getSlippage(
+        uint amountB,
+        uint amountB2
```
[the remark this answers is not in the document]
Comment (dong77) [reply packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol]: I think the naming is fine :)

File: packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamchen0510) [reply packages/loopring_v3/contracts/impl/UniswapTokenSeller.sol]: I see, I thought y and y’ were price by the function name 'getTokenToEthInputPrice', it's ok when they are amounts.

**Pull request merged**